addons/account/models/account_move.py

```diff
@@ -2046,11 +2048,11 @@ def _move_autocomplete_invoice_lines_create(self, vals_list):
                 allowed_companies = self._context.get('allowed_company_ids', journal_company.ids)
                 reordered_companies = sorted(allowed_companies, key=lambda cid: cid != journal_company.id)
                 ctx_vals['allowed_company_ids'] = reordered_companies
+            if is_invoice and 'invoice_line_ids' in vals:
+                ctx_vals['default_line_type'] = 'invl'
```

Comment:
this method is only called for invoices so you just need to deal with invl & invl_section & invl_note